src/components/network/filter-creation-panel.tsx

```diff
@@ -0,0 +1,252 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { Box, Button, Typography } from '@mui/material';
+import Grid from '@mui/material/Grid';
+import {
+    CustomFormProvider,
+    DirectoryItemSelector,
+    ElementType,
+    SelectInput,
+    TreeViewFinderNodeProps,
+} from '@gridsuite/commons-ui';
+import { yupResolver } from '@hookform/resolvers/yup';
+import yup from 'components/utils/yup-config';
+import { useForm } from 'react-hook-form';
+import { FILTER_NAME, NAME } from 'components/utils/field-constants';
+import { GridSection } from 'components/dialogs/dialogUtils';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { UniqueNameInput } from 'components/dialogs/commons/unique-name-input';
+import { useSelector } from 'react-redux';
+import {
+    equipementTypeToLabel,
+    EQUIPMENT_TYPES,
+} from '../utils/equipment-types';
+import { UUID } from 'crypto';
+import { fetchDirectoryElementPath } from '@gridsuite/commons-ui';
+import CircularProgress from '@mui/material/CircularProgress';
+import FolderOutlined from '@mui/icons-material/FolderOutlined';
+
+interface IFilterCreation {
+    [FILTER_NAME]: string | null;
+    [NAME]: string;
+    equipmentType: string | null;
```

Comment:
i think the fields are never null 
```suggestion
    [FILTER_NAME]: string;
    [NAME]: string;
    equipmentType: string;
```